frontend/src/sagas.js

```diff
@@ -830,16 +827,19 @@ export function* fetchNextProjectHistoryPageSaga({ projectId, showLoading }) {
       // After the first call we just fetch each page
       // If the offset is below 0, we have reached the last page
       // and can fetch the first events
-      offset = totalHistoryItemCount - (currentHistoryPage + 1) * historyPageSize;
-      offset = offset > 0 ? offset : 0;
+      offset = Math.max(0, totalHistoryItemCount - (currentHistoryPage + 1) * historyPageSize);
     }
 
     // If the offset is 0, we are on the last page and only need to fetch the remaining items
-    const limit = offset === 0 ? totalHistoryItemCount % historyPageSize : historyPageSize;
+    const limit =
+      offset === 0 && totalHistoryItemCount % historyPageSize !== 0
+        ? totalHistoryItemCount % historyPageSize
+        : historyPageSize;
 
     const { historyItemsCount, events } = yield callApi(api.viewProjectHistory, projectId, offset, limit);
```

Comment:
What happens if this call fails?

Reply:
When setting the browser to "offline", the loading indicator ("Loading...") is displayed and no items are fetched.